tests/unit/Lookup/DispatchingTermLookupTest.php

```diff
@@ -0,0 +1,204 @@
+<?php
+namespace Wikibase\DataModel\Services\Tests\Lookup;
+
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Services\Lookup\DispatchingTermLookup;
+use Wikibase\DataModel\Services\Lookup\TermLookup;
+use Wikibase\DataModel\Services\Lookup\UnknownForeignRepositoryException;
+use Wikimedia\Assert\ParameterAssertionException;
+
+/**
+ * @covers Wikibase\DataModel\Services\Lookup\DispatchingTermLookup
+ *
+ * @license GPL-2.0+
+ */
+class DispatchingTermLookupTest extends \PHPUnit_Framework_TestCase {
+
+	/**
+	 * @dataProvider provideInvalidForeignLookups
+	 */
+	public function testGivenInvalidForeignLookups_exceptionIsThrown( array $lookups ) {
+		$this->setExpectedException( ParameterAssertionException::class );
+		new DispatchingTermLookup( $lookups );
+	}
+
+	public function provideInvalidForeignLookups() {
+		return [
+			'no lookups given' => [[]],
+			'no lookup given for the local repository' => [
+				[ 'foo' => $this->getMock( TermLookup::class ), ]
+			],
+			'not an implementation of TermLookup given as a lookup' => [
+				[ '' => new ItemId( 'Q123' ) ],
+			],
+			'non-string keys' => [
+				[
+					'' => $this->getMock( TermLookup::class ),
+					100 => $this->getMock( TermLookup::class ),
+				],
+			],
+			'repo name containing colon' => [
+				[
+					'' => $this->getMock( TermLookup::class ),
+					'fo:oo' => $this->getMock( TermLookup::class ),
+				],
+			],
+		];
+	}
+
+	/**
+	 * @dataProvider publicMethodsProvider
+	 */
+	public function testGivenEntityFromUnknownRepo_throwsException( $method, $language ) {
+		$item = new ItemId( 'fooo:Q123' );
+		$lookup = new DispatchingTermLookup( [
+			'' => $this->getMock( TermLookup::class ),
+			'foo' => $this->getMock( TermLookup::class ),
+		] );
+
+		$this->setExpectedException( UnknownForeignRepositoryException::class );
+		$lookup->$method( $item, $language );
+	}
+
+	public function publicMethodsProvider() {
+		return [
+			[ 'getLabel', 'en' ],
+			[ 'getLabels', [ 'en', 'fr' ] ],
+			[ 'getDescription', 'en' ],
+			[ 'getDescription', [ 'en', 'de' ] ],
+		];
+	}
+
+	public function testGetLabelFromLocalRepo() {
```

Comment:
The dispatcher should not really know or care about local and foreign. It uses whatever prefix is in the ID - if the prefix is the empty string, it should just work like any other. That should be tested, but it doesn't need a separate function.